src/middleware/initiate.js

```diff
@@ -30,14 +124,42 @@ exports.initiateContextMiddleware = () => async (ctx, next) => {
     return
   }
 
+  const endpointState = await readStateByEndpointId(endpoint._id)
+
+  logger.info(`${endpoint.name} (${requestUUID}): Initiating new request`)
+
+  // initiate the property for containing all useable data points
+  ctx.state.allData = {
+    constants: endpoint.constants,
+    state: endpointState,
+    timestamps: {
+      endpointStart,
+      endpointEnd: null,
+      lookupRequests: {}
+    }
+  }
+
   // set request UUID from incoming OpenHIM header if present, else create a random UUID
   ctx.state.uuid = ctx.headers['x-openhim-transactionid']
     ? ctx.headers['x-openhim-transactionid']
-    : uuid.v4()
+    : requestUUID
   ctx.state.metaData = endpoint
 
-  logger.info(
-    `${ctx.state.metaData.name} (${ctx.state.uuid}): Initiating new request`
-  )
   await next()
+
+  try {
+    // update any specified state for this endpoint request
+    updateEndpointState(ctx, endpoint)
+  } catch (error) {
+    return handleServerError(
+      ctx,
+      'Failed to update endpoint state: ',
+      error,
+      logger
+    )
+  }
+}
+
+if (process.env.NODE_ENV === 'test') {
+  exports.updateEndpointState = updateEndpointState
```

Comment:
Instead of this export please could you use the Rewire module. Exporting private functions just feels weird :|
We've been refactoring the other middleware unit tests to be consistent

Reply:
ok cool will do :+1:

Reply:
Done as part of this commit 4a18b73